crates/node/src/lib.rs

```diff
@@ -450,7 +450,7 @@ impl Node {
 
         // sanity check
         if partition_table.num_partitions()
-            != u64::try_from(logs.num_logs()).expect("usize fits into u64")
+            != u16::try_from(logs.num_logs()).expect("usize fits into u16")
```

Comment:
`usize` probably no longer fits into `u16`. Is there another invariant that tells us that `logs.num_logs()` fits into `u16`?